src/__tests__/waitFor.test.tsx

```diff
@@ -204,6 +204,37 @@ test.each([false, true])(
   }
 );
 
+const fibonaci = (n: number): number => {
+  if (n === 0 || n === 1) {
+    return 1;
+  }
+
+  return fibonaci(n - 1) + fibonaci(n - 2);
+};
+
+test.each([false, true])(
+  'it should not depend on real time when using fake timers (legacyFakeTimers = %s)',
+  async () => {
+    jest.useFakeTimers({ legacyFakeTimers: false });
+
+    const mockErrorFn = jest.fn(() => {
+      fibonaci(30);
+      throw new Error('test');
+    });
+
+    try {
+      await waitFor(() => mockErrorFn(), {
+        timeout: 200,
+        interval: 5,
+      });
+    } catch (error) {
+      // suppress
```

Comment:
If error is expected then perhaps it's better to use `expect(...).toThrow()` as otherwise the current code works fine whether `waitFor` throws or not, which seems to be a bit fragile.

Reply:
Agreed, I use expect now